climpyrical/gridding.py

```diff
@@ -53,40 +55,45 @@ def check_input_coords(x, y):
     check_ndims(x, 1)
     check_ndims(y, 1)
 
-    if (not np.isclose(x.max(), 33.8800048828125)) or (
-        not np.isclose(x.min(), -33.8800048828125)
+    if (not np.isclose(x.max(), ds.rlon.max())) or (
+        not np.isclose(x.min(), ds.rlon.min())
```

Comment:
Why are you using `isclose()` instead of a comparison like `x.max() <= ds.rlon.max()`, as suggested by the error message (and similarly for min)?

Reply:
`x.max()` and `ds.rlon.max()` _should_ both be the same. `x` is a user-supplied array of coordinates that should be the same as in `ds.rlon`. If they are at all different, there may be significant problems downstream when it comes to mapping and gridding.

Reply:
OK, then I suggest changing the error message to reflect this. Right now it just states the looser condition.

Reply:
Ah, great, thanks for that!

Reply:
Instead of computing max and min, do you just want to compare that `x` and `ds.rlon` are approximately the same arrays, e.g., `all(isclose(a, b) for a, b in zip(x, ds.rlon))`?

Reply:
Assuming you can zip those arrays ... numpy ....

Reply:
I don't want to compare that x and ds.rlon are identical. 

>  `x` is a user-supplied array of coordinates that should be the same as in `ds.rlon`

Is true most (if not all) of the time. However, there is the possibility that a user _might_ want to use some of these functions on an interpolated grid space, in which case `x` and `y` will have the same bounds (min, max) as `ds.rlon` and `ds.rlat`, but will have 10 times as many values, and an element-wise comparison might not make sense.

Reply:
I was misleading when I said that they can't "at all be different" - what I mean is "their bounds cannot be different".

Reply:
I wondered if that was the case. Just checking all the assumptions. It's very useful to state these in the documentation, too, for others working with the code (or yourself in future).

Reply:
Done. Thanks.